MAINT: iid=True in `cov.py` is deprecated

On latest `sklearn` master I get this in tests:
```
mne/cov.py:991: in _compute_covariance_auto
    gs.fit(data_[:, picks])
...
E           DeprecationWarning: The parameter 'iid' is deprecated in 0.22 and will be removed in 0.24.
```
Which is really from `cov.py` -- here showing where `gs` is instantiated and used:
```
            elif method_ == 'shrunk':
                ...
                gs = GridSearchCV(ShrunkCovariance(**mp),
                                  tuned_parameters, cv=cv, iid=True)
                for ch_type, picks in sub_picks_list:
                    gs.fit(data_[:, picks])
                    shrinkages.append((ch_type, gs.best_estimator_.shrinkage,
                                       picks))
```
I was hoping this would be an easy `**kwargs`-based-on-version fix, but it looks like `iid` is going away -- from the [sklearn docs](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html):
> iid : boolean, default=’warn’
>    If True, return the average score across folds, weighted by the number of samples in each test set. In this case, the data is assumed to be identically distributed across the folds, and the loss minimized is the total loss per sample, and not the mean loss across the folds. If False, return the average score across folds. Default is True, but will change to False in version 0.22, to correspond to the standard definition of cross-validation.
>    Changed in version 0.20: Parameter iid will change from True to False by default in version 0.22, and will be removed in 0.24.

@agramfort what do we do here? Is `iid=True` important, or is setting it to `iid=False` okay?

--- a/mne/cov.py
+++ b/mne/cov.py
@@ -5,6 +5,7 @@
 # License: BSD (3-clause)
 
 from copy import deepcopy
+from distutils.version import LooseVersion
 import itertools as itt
 from math import log
 import os
@@ -898,6 +899,14 @@ def _eigvec_subspace(eig, eigvec, mask):
     return eig, eigvec
 
 
+def _get_iid_kwargs():
+    import sklearn
+    kwargs = dict()
+    if LooseVersion(sklearn.__version__) < LooseVersion('0.22'):
+        kwargs['iid'] = False
+    return kwargs
+
+
 def _compute_covariance_auto(data, method, info, method_params, cv,
                              scalings, n_jobs, stop_early, picks_list, rank):
     """Compute covariance auto mode."""
@@ -986,7 +995,7 @@ def _compute_covariance_auto(data, method, info, method_params, cv,
                 tuned_parameters = [{'shrinkage': shrinkage}]
                 shrinkages = []
                 gs = GridSearchCV(ShrunkCovariance(**mp),
-                                  tuned_parameters, cv=cv, iid=True)
+                                  tuned_parameters, cv=cv, **_get_iid_kwargs())
                 for ch_type, picks in sub_picks_list:
                     gs.fit(data_[:, picks])
                     shrinkages.append((ch_type, gs.best_estimator_.shrinkage,

